glusterfs/datadog_checks/glusterfs/check.py

```diff
@@ -0,0 +1,193 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import json
+
+try:
+    from json import JSONDecodeError
+except ImportError:
+    from simplejson import JSONDecodeError
+
+import os
+import shlex
+from typing import Any
+
+from six import iteritems
+
+from datadog_checks.base import AgentCheck, ConfigurationError
+from datadog_checks.base.config import is_affirmative
+from datadog_checks.base.utils.subprocess_output import get_subprocess_output
+
+from .metrics import BRICK_STATS, CLUSTER_STATS, PARSE_METRICS, VOL_SUBVOL_STATS, VOLUME_STATS
+
+GLUSTER_VERSION = 'glfs_version'
+CLUSTER_STATUS = 'cluster_status'
+
+GSTATUS_PATH = '/opt/datadog-agent/embedded/sbin/gstatus'
+
+
+class GlusterfsCheck(AgentCheck):
+    __NAMESPACE__ = 'glusterfs'
+
+    CLUSTER_SC = "cluster.health"
+    VOLUME_SC = "volume.health"
+    BRICK_SC = "brick.health"
+
+    def __init__(self, name, init_config, instances):
+        # type: (*Any, **Any) -> None
+        super(GlusterfsCheck, self).__init__(name, init_config, instances)
+        self._tags = self.instance.get('tags', [])
+
+        # Check if customer set gstatus path
+        if init_config.get('gstatus_path'):
+            self.gstatus_cmd = init_config.get('gstatus_path')
+        else:
+            if os.path.exists(GSTATUS_PATH):
+                self.gstatus_cmd = GSTATUS_PATH
+            else:
+                raise ConfigurationError(
+                    'Glusterfs check requires `gstatus` to be installed or set the path to the installed version.'
+                )
+        self.log.debug("Using gstatus path `%s`", self.gstatus_cmd)
+        self.use_sudo = is_affirmative(self.instance.get('use_sudo', False))
+
+
+    def check(self, _):
+        if self.use_sudo:
+            test_sudo, err, rcode = get_subprocess_output(['sudo', '-ln', self.gstatus_cmd], self.log)
+            if rcode != 0:
+                raise Exception('The dd-agent user does not have sudo access: %s', err or test_sudo)
+            gluster_args = 'sudo {}'.format(self.gstatus_cmd)
+        else:
+            gluster_args = self.gstatus_cmd
```

Comment:
I understand that your want the user to explicitly ask for sudo permission, but if we know that glusterfs requires sudo we could fail at that line directly right? This would allow for a much better error message.

Reply:
So the default setup of GlusterFS does require sudo/root access. However, there are workarounds that support non-sudo access that requires additional configuration. I'm going to still provide an option to disable sudo for that possibility but change the default to `true`. WDYT?

Reply:
Ok I didn't understand that you could workaround the sudo requirement.
Sounds good 👍